packages/contracts/test/ubiquistick/UbiquiStick.t.sol

```diff
@@ -0,0 +1,175 @@
+        bannedOperator = 0x2B2e8cDA09bBA9660dCA5cB6233787738Ad68329; // SudoSwap LSSVMPairRouter
+        allowedOperator = 0x4feE7B061C97C9c496b01DbcE9CDb10c02f0a0Be; // Rarible NFT Transfer Proxy (for Approvals)
+    }
+
+    // Mint
+    function testMint() public {
+        assertEq(ust.balanceOf(mintTo), 0);
+        ust.safeMint(mintTo);
+        assertEq(ust.balanceOf(mintTo), 1);
+        assertEq(ust.tokenIdNext(), 2);
+    }
+
+    // APPROVALS
+    function testApproveForAllowedOperator() public {
+        // mint NFT to vitalik's address
+        ust.safeMint(mintTo);
+
+        // Approval for an allowed operator should work
+        vm.startPrank(mintTo);
+        vm.expectEmit(true, true, false, true);
+        emit Approval(mintTo, allowedOperator, 1);
+        ust.approve(allowedOperator, 1);
+        vm.stopPrank();
+    }
+
+    function testApproveForBannedOperator() public {
+        // mint NFT to vitalik's address
+        ust.safeMint(mintTo);
+
+        // Approval for banned operator should fail with the AddressFiltered(address filtered) error
+        vm.expectRevert(abi.encodeWithSelector(OperatorFilterRegistryErrorsAndEvents.AddressFiltered.selector, bannedOperator));
+        vm.prank(mintTo);
+        ust.approve(bannedOperator, 1);
+    }
+
+    function testSetApprovalsAllForAllowedOperator() public {
+        // mint NFT to vitalik's address
+        ust.safeMint(mintTo);
+
+        // Approval for an allowed operator should work
+        vm.startPrank(mintTo);
+        vm.expectEmit(true, true, false, true);
+        emit ApprovalForAll(mintTo, allowedOperator, true);
+        ust.setApprovalForAll(allowedOperator, true);
+        vm.stopPrank();
+    }
+
+    function testSetApprovalsAllForBannedOperator() public {
+        // mint NFT to vitalik's address
+        ust.safeMint(mintTo);
+
+        // Approval for banned operator should fail with the AddressFiltered(address filtered) error
+        vm.expectRevert(abi.encodeWithSelector(OperatorFilterRegistryErrorsAndEvents.AddressFiltered.selector, bannedOperator));
+        vm.prank(mintTo);
+        ust.setApprovalForAll(bannedOperator, true);
+    }
+
+    // TRANSFERS
+    function testTransferFromForBannedOperator() public {
+        // mint NFT to vitalik's address
+        ust.safeMint(mintTo);
+
+        // Approval for banned operator should fail with the AddressFiltered(address filtered) error
+        vm.expectRevert(abi.encodeWithSelector(OperatorFilterRegistryErrorsAndEvents.AddressFiltered.selector, bannedOperator));
+        vm.prank(mintTo);
+        ust.approve(bannedOperator, 1);
+
+        // transferFrom vitalik's address to banned operator should also fail with the AddressFiltered(address filtered) error
+        vm.expectRevert(abi.encodeWithSelector(OperatorFilterRegistryErrorsAndEvents.AddressFiltered.selector, bannedOperator));
+        vm.prank(bannedOperator);
+        ust.transferFrom(mintTo, transferTo, 1);
```

Comment:
Lets add a check for the result of transfer like this `assertEq(afterBalance - beforeBalance, 1)` and transfer event as well